### PR TITLE
MUSARK-993 - feature/missing sample fields

### PR DESCRIPTION
**Pull request by ingarabr**
- treatment
- residualMaterial
- description
- externalId.source

### CODE REVIEW

File: service_management/app/models/analysis/ResidualMaterials.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+object ResidualMaterials {
+
+  sealed trait ResidualMaterial {
+    val identity: Int
+  }
+
+  object ResidualMaterial {
+    def fromInt(i: Int): Option[ResidualMaterial] = i match {
+      case NotSpecified.identity        => Some(NotSpecified)
```
Comment (kpmeen) [inline service_management/app/models/analysis/ResidualMaterials.scala]: It's preferable give `identity` a different name. So there's no ambiguity with `Predef.identity`.

File: service_management/app/repositories/analysis/dao/Tables.scala [hunk 2]
```diff
@@ -304,7 +309,7 @@ trait Tables extends HasDatabaseConfigProvider[JdbcProfile] with ColumnTypeMappe
       maybeTuple: Option[ResultRow]
   ): Option[AnalysisResult] =
     maybeTuple.flatMap(fromResultRow)
-
+  //todo add externalIdSource, treatment, residualMaterial, description
```
Comment (kpmeen) [inline service_management/app/repositories/analysis/dao/Tables.scala]: Is this comment still valid?

**Review by kpmeen — COMMENTED** [review]
Just one mine thing

File: service_management/app/models/analysis/ResidualMaterials.scala (hunk 1 above)
Comment (ingarabr) [inline service_management/app/models/analysis/ResidualMaterials.scala]: nope, I'll remove it

File: service_management/app/repositories/analysis/dao/Tables.scala (hunk 2 above)
Comment (ingarabr) [reply service_management/app/repositories/analysis/dao/Tables.scala]: nope, I'll remove it

Comment (steinaol) [reply service_management/app/repositories/analysis/dao/Tables.scala]: I have a question concerning different actors (persons) with different roles and dates connected to a sample (i.e. the sampleCreation [Prøveuttak]). From frontend we have an array with person-, role- and date objects and. How should we send this to backend? An array called Persons, where each entry is an object with type { name: string, role: string, date: string}?

Comment (ingarabr) [reply service_management/app/repositories/analysis/dao/Tables.scala]: Flatten them out based on there roles.

**Review by kpmeen — APPROVED** [review]

**Pull request merged**